src/User32.Desktop/User32.cs

```diff
@@ -57,7 +56,61 @@ public static partial class User32
         public static readonly IntPtr HBMMENU_SYSTEM = new IntPtr(1);
 
-        public delegate int WindowsHookDelegate(int code, IntPtr wParam, IntPtr lParam);
+        /// <summary>
+        /// An application-defined or library-defined callback function used with the <see cref="SetWindowsHookEx(WindowsHookType, IntPtr, IntPtr, int)"/> function.
+        /// This is a generic function to Hook callbacks. For specific callback functions see CallWndDelegate, CallWndRetDelegate, CBTDelegate, DebugDelegate, ForegroundIdleDelegate,
+        /// GetMsgDelegate, JournalPlaybackDelegate, JournalRecordDelegate, KeyboardDelegate, LowLevelKeyboardDelegate, MessageDelegate, MouseDelegate, LowLevelMouseDelegate,
+        /// ShellDelegate and SysMsgDelegate.
+        /// </summary>
+        /// <param name="nCode">An action code for the callback. Can be used to indicate if the hook procedure must process the message or not.</param>
+        /// <param name="wParam">First message parameter</param>
+        /// <param name="lParam">Second message paramater</param>
+        /// <returns>
+        /// An LRESULT. Ususally if nCode is less than zero, the hook procedure must return the value returned by CallNextHookEx.
+        /// If nCode is greater than or equal to zero, it is highly recommended that you call CallNextHookEx and return the value it returns;
+        /// otherwise, other applications that have installed hooks will not receive hook notifications and may behave incorrectly as a result.
+        /// If the hook procedure does not call CallNextHookEx, the return value should be zero.
+        /// </returns>
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        public delegate int WindowsHookDelegate(int nCode, IntPtr wParam, IntPtr lParam);
+
+        /// <summary>
+        /// An application-defined callback function used with the <see cref="EnumWindowStations(EnumWindowStationDelegate, IntPtr)"/> function.
+        /// </summary>
+        /// <param name="lpszWindowStation">The name of the window station.</param>
+        /// <param name="lParam">An application-defined value specified in the <see cref="EnumWindowStations(EnumWindowStationDelegate, IntPtr)"/> function.</param>
+        /// <returns>To continue enumeration, the callback function must return TRUE (non-zero value). To stop enumeration, it must return FALSE (0).</returns>
+        /// <remarks>
+        /// An application must register this callback function by passing its address to <see cref="EnumWindowStations(EnumWindowStationDelegate, IntPtr)"/>.
+        /// The callback function can call SetLastError to set an error code for the caller to retrieve by calling GetLastError.
+        /// </remarks>
+        [UnmanagedFunctionPointer(CallingConvention.StdCall, CharSet = CharSet.Ansi)]
+        public delegate int EnumWindowStationDelegate(string lpszWindowStation, IntPtr lParam);
```

Comment:
We could use that naming, but I looked for other delegates and callbacks on the project and followed what we had for WindowHooks, it was changed from WindowsHookProc to WindowsHookDelegate also. I would like most to have the same names as we have on MSDN and would change the WindowsHook if it is OK for everyone, but that might break backwards compatibility.

About the return, this seems to be a tricky thing. If you look on the remarks of the EnumWIndowsStation method you will see that it does both. If it receives zero it stops enumerating understanding an error happened or that the callback simply don't want to continue enumerating, but on any other cases the non-zero result should will be propagated to the EnumWindowStation function as its own return. Tricky but interesting behavior. It enables the developer to have something doing the callback on a completely different part of the code but still being able to send an integer encoded result code to the caller. I found it to be so cool that I tried to maintain the behavior.